addons/mail/models/mail_template.py

```diff
@@ -146,12 +146,21 @@ def _fix_attachment_ownership(self):
             record.attachment_ids.write({'res_model': record._name, 'res_id': record.id})
         return self
 
+    def _check_abstract_models(self, vals_list):
+        model_names = self.sudo().env['ir.model'].browse([
+            vals['model_id'] for vals in vals_list if vals.get('model_id')
+        ]).mapped('model')
```

Comment:
Didn't know you could just pass iterators, good to know. Though since we'll iterate over the whole thing either way it probably doesn't change much, if anything (?)